clients/auth/auth.go

```diff
@@ -0,0 +1,431 @@
+			Token:       a.GetToken(),
+			Status:      a.GetStatus(),
+			UserName:    a.GetUser(),
+			UserID:      a.GetUserID(),
+			Permissions: ps,
+		})
+	}
+
+	return c.printAuth(printParams{tokens: tokens})
+}
+
+func (c Client) findAuthorization(ctx context.Context, authID string) error {
+	a, err := c.GetAuthorizationsID(ctx, authID).Execute()
+	if err != nil {
+		return fmt.Errorf("could not find authorization with ID %q: %w", authID, err)
+	}
+
+	ps := make([]string, 0, len(a.GetPermissions()))
+	for _, p := range a.GetPermissions() {
+		ps = append(ps, permString(p))
+	}
+
+	return c.printAuth(printParams{
+		token: &token{
+			ID:          a.GetId(),
+			Description: a.GetDescription(),
+			Token:       a.GetToken(),
+			Status:      a.GetStatus(),
+			UserName:    a.GetUser(),
+			UserID:      a.GetUserID(),
+			Permissions: ps,
+		},
+	})
+}
+
+func (c Client) SetActive(ctx context.Context, authID string, active bool) error {
+
+	// check if auth exists first for better error logging
+	if _, err := c.GetAuthorizationsID(ctx, authID).Execute(); err != nil {
+		return fmt.Errorf("could not find auth with ID %q: %w", authID, err)
+	}
+
+	var status string
+	if active {
+		status = "active"
+	} else {
+		status = "inactive"
+	}
+	a, err := c.PatchAuthorizationsID(ctx, authID).
+		AuthorizationUpdateRequest(api.AuthorizationUpdateRequest{Status: &status}).
+		Execute()
+	if err != nil {
+		return fmt.Errorf("could not update status of auth with ID %q: %w", authID, err)
+	}
+
+	ps := make([]string, 0, len(a.GetPermissions()))
+	for _, p := range a.GetPermissions() {
+		ps = append(ps, permString(p))
+	}
+
+	return c.printAuth(printParams{
+		token: &token{
+			ID:          a.GetId(),
+			Description: a.GetDescription(),
+			Token:       a.GetToken(),
+			Status:      a.GetStatus(),
+			UserName:    a.GetUser(),
+			UserID:      a.GetUserID(),
+			Permissions: ps,
+		},
+	})
+}
+
+func (c Client) printAuth(opts printParams) error {
+	if c.PrintAsJSON {
+		var v interface{}
+		if opts.token != nil {
+			v = opts.token
+		} else {
+			v = opts.tokens
+		}
+		return c.PrintJSON(v)
+	}
+
+	headers := []string{
+		"ID",
+		"Description",
+		"Token",
+		"User Name",
+		"User ID",
+		"Permissions",
+	}
+	if opts.deleted {
+		headers = append(headers, "Deleted")
+	}
+
+	if opts.token != nil {
+		opts.tokens = append(opts.tokens, *opts.token)
+	}
+
+	var rows []map[string]interface{}
+	for _, t := range opts.tokens {
+		row := map[string]interface{}{
+			"ID":          t.ID,
+			"Description": t.Description,
+			"Token":       t.Token,
+			"User Name":   t.UserName,
+			"User ID":     t.UserID,
+			"Permissions": t.Permissions,
+		}
+		if opts.deleted {
+			row["Deleted"] = true
+		}
+		rows = append(rows, row)
+	}
+	return c.PrintTable(headers, rows...)
+}
+
+func permString(p api.Permission) string {
+	ret := p.GetAction() + ":"
+	r := p.GetResource()
+
+	if r.GetOrgID() != "" {
+		ret += "orgs/" + r.GetOrgID()
+	}
+	ret += "/" + r.GetType()
```

Comment:
If `GetOrgID()` returns empty string, we'll end up with a permission string like `action:/type`. I think all our permissions are org-scoped in practice at the moment, but it'd be nice to future-proof.

Reply:
Also, I think this is copied from the v1 auth code. Can we centralize? You could create something like `api/permission_string.go` and then write it there as a method on `Permission`